lib/ansible/executor/action_write_locks.py

```diff
@@ -37,7 +37,7 @@
     # These plugins are called directly by action plugins (not going through
     # a strategy).  We precreate them here as an optimization
     mods = set(p['name'] for p in Facts.PKG_MGRS)
-    mods.update(('copy', 'file', 'setup', 'slurp', 'stat'))
+    mods.update(('copy', 'file', 'setup', 'slurp', 'stat', 'gather_facts'))
```

Comment:
You can remove setup here if it no longer gets called directly from an action plugin